overlord/snapstate/storehelpers.go

```diff
@@ -44,7 +54,17 @@ func updateInfo(st *state.State, snapst *SnapState, channel string, ignoreValida
 	}
 
 	if curInfo.SnapID == "" { // covers also trymode
-		return nil, store.ErrLocalSnap
+		if !amend {
+			return nil, store.ErrLocalSnap
+		}
+
+		// in amend mode we need to move to the store rev
+		id, err := snapNameToSnapIDFromStore(st, curInfo.Name(), user)
+		if err != nil {
+			return nil, fmt.Errorf("cannot get snap-id for %q: %v", curInfo.Name(), err)
+		}
+		curInfo.SnapID = id
+		curInfo.Revision = snap.R(1) // we really have no idea
```

Comment:
Lying about the revision here feels dangerous in the context of epochs. I think we should omit this information when we don't know. Will need some discussion with the store folks to make sure this works correctly on the other end.

@pedronis Can you close the loop here?  Also a good chance to talk about epochs.. we should send the epoch so the store has a clue about what to return.